src/Microsoft.Health.Fhir.Liquid.Converter/Filters/SegmentFilters.cs

```diff
@@ -128,5 +128,50 @@ public static bool HasSegments(Hl7v2Data hl7v2Data, string segmentIdContent)
 
             return result;
         }
+
+        public static List<Hl7v2Data> SliceDataBySegments(Hl7v2Data hl7v2Data, List<Hl7v2Segment> segmentList, Hl7v2Segment endSegment = null)
```

Comment:
Just come up with a strange case: what will happen if `endSegment` is one of the segments in `segmentList`, e.g., the first one, the second one?

Reply:
need some functional doc

Reply:
If `endsegment` is encountered, it should be stopped even though there are elements not traversed in the `segmentList`. If user don't want to stop it, user should select an appropriate `endsegment`. We can note it in the document.

Reply:
Can we add a unit test about this case?

Reply:
Sure~